lib/statsample-glm/glm/mle/base.rb

```diff
@@ -109,11 +109,11 @@ def newton_raphson(x,y, start_values=nil)
             @iterations = i + 1
 
             h = second_derivative(x,y,parameters)
-            if h.singular?
+            if h.det() == 0
               raise "Hessian is singular!"
             end
             fd = first_derivative(x,y,parameters)
-            parameters = parameters - (h.inverse * (fd))
+            parameters = parameters - (h.invert.dot fd)
```

Comment:
Don't use `invert` when you can use `solve`.
`h.invert.dot fd` is equivalent to `h.solve fd`.